axiom/nr_php_packages.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef nr_php_packages_HDR
+#define nr_php_packages_HDR
+
+#include "nr_log_event.h"
+#include "util_random.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+
+typedef struct _nr_php_package_t {
+  char* package_name;
+  char* package_version;
+} nr_php_package_t;
+
+typedef nr_hashmap_t nr_php_packages_t;
+
+/*
+ * Purpose : Create a new php package
+ *
+ * Params  : 1. Package name
+ *           2. Package version
+ * 
+ * Returns : A php package that has a name and version. If nr_php_packages_add_package()
+ *           is not called, then it must be freed by nr_php_package_destroy()
+ */
+extern nr_php_package_t* nr_php_package_create(char* name, char* version);
+
+/*
+ * Purpose : Destroy/free php package
+ *
+ * Params  : The php package to free
+ * 
+ * Returns : Nothing, it is void
+ */
+extern void nr_php_package_destroy(nr_php_package_t* p);
+
+/*
+ * Purpose : Add new php package to hashmap. If the hashmap does not exist and this is the first package
+ *           being added, then a new hashmap will be created by calling nr_hashmap_create(). This function
+ *           also frees the package being added, so the user does not need to call nr_php_packages_destroy_package()
```

Comment:
Could you better describe what happens if the packages hashmap already contains a package with the same name? Could you add a test for that behavior?

Reply:
I have just pushed a commit that will handle the case where a package is added with the same name. If a new package is added with the same name but a different value, then the newer value will be kept and the older value will be freed. If a package is added with the same name and value, then no action will happen. [f5f2d0b](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/f5f2d0b1802cd6ab3c1f71a62c28150008a223fa)